### PR TITLE
Fix example app for newer CocoaPods & update for Xcode 8

### PR DESCRIPTION
**Pull request by SSheldon**
If you `pod install` using a newer version of CocoaPods, first you get this error:
```
[!] Unable to integrate the following embedded targets with their respective host targets (a host target is a "parent" target which embeds a "child" target like a framework or extension):

- YelpAPITests (iOS 9.3) and YelpAPI (iOS 8.0) do not use the same platform.
```

The fix for that is to change the deployment target of the tests to match the framework itself. But even then the example app crashes on launch with:
```
dyld: Library not loaded: @rpath/OMGHTTPURLRQ.framework/OMGHTTPURLRQ
  Referenced from: /Users/ssheldon/Library/Developer/Xcode/DerivedData/YelpAPI-glukeheqyvsrdfdeyahpzjtdtakh/Build/Products/Debug-iphonesimulator/YelpAPI.framework/YelpAPI
  Reason: image not found
```

One hint as to what's wrong: if you remove the test target from the podfile, cocoapods 1.1.0 gives you this warning (added in CocoaPods/CocoaPods#5733, thanks @benasher44!):
```
[!] The Podfile contains framework targets, for which the Podfile does not contain host targets (targets which embed the framework).
If this project is for doing framework development, you can ignore this message. Otherwise, add a target to the Podfile that embeds these frameworks to make this message go away (e.g. a test target).
```

So, it sounds like we need to have cocoapods manage the example app for this to work. However, I found that cocoapods has trouble with this while the example app is in a different project from the framework. So, to fix this I also needed to move the example app into the main project. This is a win anyways, because now we don't have two YelpAPI.xcodeproj and people won't be confused about which to open.

Testing done: got travis tests passing, verified that the example app works

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 language: objective-c
-osx_image: xcode7.1
+osx_image: xcode8
 xcode_workspace: YelpAPI.xcworkspace
 xcode_scheme: YelpAPI
-xcode_sdk: iphonesimulator9.1
+xcode_sdk: iphonesimulator
```
Comment (thibaudrobelain) [inline .travis.yml]: does this need to be pinned to 10.0?

**Comment by thibaudrobelain** [comment]
@SSheldon: Just need an answer for my small comment above and I'm good with this!

File: .travis.yml (hunk 1 above)
Comment (SSheldon) [inline .travis.yml]: If you don't specify a version number, like this, it uses the latest version of the SDK. That matches what's in our project file (where we ask to build against the latest available version). basher made this change to YLTableView in Yelp/YLTableView#16, and I didn't want to keep updating the version number so it seemed like a good idea 😛

**Comment by thibaudrobelain** [comment]
Shipit!

**Pull request merged**